pandas/io/tests/sas/test_sas7bdat.py

```diff
@@ -44,27 +44,46 @@ def test_from_buffer(self):
             df0 = self.data[j]
             for k in self.test_ix[j]:
                 fname = os.path.join(self.dirpath, "test%d.sas7bdat" % k)
-                with open(fname, 'rb') as f:
-                    byts = f.read()
-                buf = io.BytesIO(byts)
-                df = pd.read_sas(buf, format="sas7bdat", encoding='utf-8')
```

Comment:
I think it is more logical to keep the buffer reading here, as this test is called "test_from_buffer" is I suppose is exactly testing this (and  then maybe use plain reading from file in "test_iterator_read_too_much")